force-app/main/default/classes/DonationHistoryController.cls

```diff
@@ -47,15 +47,19 @@ public with sharing class DonationHistoryController {
         @AuraEnabled public String creditType;
         @AuraEnabled public Decimal amount;
         @AuraEnabled public String paymentMethod;
-        public DonationHistoryView(Opportunity ocr){
+
+        public DonationHistoryView(Opportunity ocr, String paymentMethod){
             this.id = ocr.Id;
             this.name = ocr.Name;
             this.closeDate = ocr.CloseDate;
             this.amount = ocr.Amount;
             this.creditType = System.Label.CRLP_HardCredit;
-            this.paymentMethod = 'Cash';
+            this.paymentMethod = paymentMethod;
         }
     }
+    
+    private static String EMPTY_PAYMENT_METHOD = '-';
```

Comment:
This should be BEFORE the constructor and has to be defined as final too private static final String...